server/utils.go

```diff
@@ -397,6 +399,41 @@ func versionToString(vers int) string {
 	return str
 }
 
+// rewriteToken attempts to match the original token
+// against the email, telephone number or login patterns.
+// On success, prepends the token with the corresponding prefix.
+func rewriteToken(orig string) string {
+	if orig == "" || strings.HasPrefix(orig, "basic:") ||
+			strings.HasPrefix(orig, "email:") || strings.HasPrefix(orig, "tel:") {
+		return orig
+	}
+	// Is it email?
+	if emailRegexp.MatchString(orig) {
+		return "email:" + orig
+	}
+	// TODO: pass region as a param.
```

Comment:
I think it should be 

1. Use whatever client has provided
2. Try to infer from client's phone number or location.
3. Use value from `.conf` file.
4. Fallback to `US` as a last resort.

Reply:
Added to the TODO.